### ISSUE TITLE
[BUG] Crashes when using Tic-tok or 3d tests

### ISSUE THREAD

**Issue by TaPo4eK228** [MustardChef/WSABuilds#459]
### Describe the bug

Any application that requires a bit of complex graphical rendering just crashes.

https://github.com/user-attachments/assets/2033b5fe-1047-4622-97ec-e4ad58dc12ca

### Steps to reproduce the issue

Run tik tok or any 3d test

### Expected behaviour

just working application

### Downloaded Build Of WSA

[WSABuilds LTS Build #3 (03/06/24) for Windows 11 x64](https://github.com/MustardChef/WSABuilds/releases/tag/Windows_11_2311.40000.5.0_LTS_3)
[WSA_2311.40000.5.0_x64_Release-Nightly-GApps-13.0-NoAmazon.7z](https://github.com/MustardChef/WSABuilds/releases/download/Windows_11_2311.40000.5.0_LTS_3/WSA_2311.40000.5.0_x64_Release-Nightly-GApps-13.0-NoAmazon.7z)

### Windows build number

Win 11, 23H2 (OC 22631.4317)

### PC Specification

- AMD Ryzen 7 5700x 
- RAM: Kingston 16GB 3200 MHz
- SSD Adata Legeng 960, 1TB
- MSI RTX 3060 (12GB)
- MB: MSI b550 Pro-VDH

### Additional context

It's probably worth starting with the fact that I saw the inscription about nvidia graphics cards and, in theory, I have to suffer with it myself, but still I ask for help. 
I can also add that the point about the Vulcan driver looks strange (perhaps it is an inaccuracy of translation into my language).
![image](https://github.com/user-attachments/assets/268fb469-f1b3-4b3c-be77-30ed153012ba)
Changing the item in the screenshot does not change anything

A little more about Vulcan using the example of 3DMark:
![image](https://github.com/user-attachments/assets/d95b2439-39cf-497a-b695-573bae4271ce)
![image](https://github.com/user-attachments/assets/97006fa9-8b07-4c81-a6eb-8a055df52252)
It looks wrong too.

I can say that regular applications such as YouTube or Google play and third-party ones work well, unlike Tik Tok. I decided to address this problem because I remember that it worked on my last computer, even though it took 15 minutes of work to crash. There was less RAM and there was an intel hd graphics 630, besides a weak graphics card. I don't remember exactly what was selected there, but I'll just say it all for information.

If you need any additional information to solve the problem, pls write, I usually respond quickly.

**Comment by MustardChef** [comment]
To answer your questions regarding Vulkan on WSA: There are two options that you can choose from.

- ``System Default``: Uses the Vulkan driver provided by your GPU vendor. In this case it would be Nvidia.

- ``D3D12``: Uses Vulkan on D3D12 provided by Mesa Dozen. This is typically slower than using the ``System Default`` option as it was still in beta/testing phase.

As to the reason why 3DMark states that there are missing Vulkan features. This is because WSA only supports Vulkan 1.1, and therefore is incompatible with the benchmark you are attempting to run.

Try enabling Vulkan by setting it to ``System Default`` and restart TikTok to see if this helps.

**Comment by TaPo4eK228** [comment]
> Try enabling Vulkan by setting it to `System Default` and restart WSA to see if this helps.

Yes, it solved the problem with 3DMark, but it didn't solve the problem with tik tok. It starts, sometimes even shows 1 video after a minute of waiting, but when i am scrolling or pressing buttons, it also freezes and then crashes

**Comment by TaPo4eK228** [comment]
@MustardChef Do you have any ideas how to fix tik tok?

**Comment by Schefflin** [comment]
> To answer your questions regarding Vulkan on WSA: There are two options that you can choose from.
> 
> * `System Default`: Uses the Vulkan driver provided by your GPU vendor. In this case it would be Nvidia.
> * `D3D12`: Uses Vulkan on D3D12 provided by Mesa Dozen. This is typically slower than using the `System Default` option as it was still in beta/testing phase.
> 
> As to the reason why 3DMark states that there are missing Vulkan features. This is because WSA only supports Vulkan 1.1, and therefore is incompatible with the benchmark you are attempting to run.
> 
> Try enabling Vulkan by setting it to `System Default` and restart WSA to see if this helps.

@MustardChef I can't use default vulkan with Intel Arc Xe graphics but d3d12, is it normal?

**Comment by TehCrazyCat** [comment]
I am having the same issue with TikTok, it crashes when attempting to render anything. It lagged before but it wasn't that bad. All other social apps like Instagram behave normally, even after prolonged usage.

Already tried modifying Vulkan render and RAM. Any solution?

I have Non-LTCS 2407.40000.0.0, w/ GApps, No Amazon

My PC specs:
Ryzen 5 2600
AMD RX 570
32 GB RAM

**Comment by TaPo4eK228** [comment]
in version 2407.40000.4.0_x64 is still the same problem

**Comment by WutIsHummus** [comment]
my tiktok wont even open

![Image](https://github.com/user-attachments/assets/95a4b70b-b362-49a1-a971-7bfa2ee1c275)

**Comment by TaPo4eK228** [comment]
I do not know why, but now everything is working for me on version 2407.40000.0.0.

**Comment by JakeDansker** [comment]
Is it even possible to update to the newest WSA with Magisk version since its been deprecated? I'm on a much older version which I think could be causing the issue.

**Comment by NebularNerd** [comment]
Just updated to WSA_2407.40000.4.0_x64_Release-Nightly-with-magisk-29.0.29000.-stable-GApps-13.0 on Win 11. TikTok now starts for me but either crashes immediately or as you try to login it bugs out. I've tried Force Close/Delete Data, Uninstall/Reinstall, Checked Vulkan set to System Default/D3D12. No joy.

**Comment by MustardChef** [comment]
> Just updated to WSA_2407.40000.4.0_x64_Release-Nightly-with-magisk-29.0.29000.-stable-GApps-13.0 on Win 11. TikTok now starts for me but either crashes immediately or as you try to login it bugs out. I've tried Force Close/Delete Data, Uninstall/Reinstall, Checked Vulkan set to System Default/D3D12. No joy.

@NebularNerd Known issue. Please wait for the next LTS build which should hopefully solve the issue.

**Comment by johnwyatt041-commits** [comment]
> > Just updated to WSA_2407.40000.4.0_x64_Release-Nightly-with-magisk-29.0.29000.-stable-GApps-13.0 on Win 11. TikTok now starts for me but either crashes immediately or as you try to login it bugs out. I've tried Force Close/Delete Data, Uninstall/Reinstall, Checked Vulkan set to System Default/D3D12. No joy.
> 
> [@NebularNerd](https://github.com/NebularNerd) Known issue. Please wait for the next LTS build which should hopefully solve the issue.

You can try this TikTok version. It works for my WSA tiktok-41-8-15

**Comment by johnwyatt041-commits** [comment]
> > > Just updated to WSA_2407.40000.4.0_x64_Release-Nightly-with-magisk-29.0.29000.-stable-GApps-13.0 on Win 11. TikTok now starts for me but either crashes immediately or as you try to login it bugs out. I've tried Force Close/Delete Data, Uninstall/Reinstall, Checked Vulkan set to System Default/D3D12. No joy.
> > 
> > 
> > [@NebularNerd](https://github.com/NebularNerd) Known issue. Please wait for the next LTS build which should hopefully solve the issue.
> 
> You can try this TikTok version. It works for my WSA tiktok-41-8-15

It works but sometimes the app gets stuck.

**Comment by johnwyatt041-commits** [comment]
> > > Just updated to WSA_2407.40000.4.0_x64_Release-Nightly-with-magisk-29.0.29000.-stable-GApps-13.0 on Win 11. TikTok now starts for me but either crashes immediately or as you try to login it bugs out. I've tried Force Close/Delete Data, Uninstall/Reinstall, Checked Vulkan set to System Default/D3D12. No joy.
> > 
> > 
> > [@NebularNerd](https://github.com/NebularNerd) Known issue. Please wait for the next LTS build which should hopefully solve the issue.
> 
> You can try this TikTok version. It works for my WSA tiktok-41-8-15

sorry not this version. It should be 25.2.4. This version works really smooth

**Comment by darpna** [comment]
Hi, is there any updates on this? @MustardChef @TaPo4eK228

**Comment by TaPo4eK228** [comment]
It still doesn't work for me.

**Comment by MustardChef** [comment]
> Hi, is there any updates on this? [@MustardChef](https://github.com/MustardChef) [@TaPo4eK228](https://github.com/TaPo4eK228)

This issue should be fixed in the next release.
